actionpack/test/controller/rate_limiting_test.rb

```diff
@@ -1,17 +1,9 @@
 # frozen_string_literal: true
 
 require "abstract_unit"
-require "kredis"
-
-Kredis.configurator = Class.new do
-  def config_for(name) { db: "2" } end
-  def root() Pathname.new(Dir.pwd) end
-end.new
-
-# Enable Kredis logging
-# ActiveSupport::LogSubscriber.logger = ActiveSupport::Logger.new(STDOUT)
 
 class RateLimitedController < ActionController::Base
+  self.cache_store = ActiveSupport::Cache::MemoryStore.new
   rate_limit to: 2, within: 2.seconds, by: -> { Thread.current[:redis_test_seggregation] }, only: :limited_to_two
```

Comment:
Since we're no longer using a shared store, I think we can drop `:by` here and on line 13 (and the assignment in `setup`).

Reply:
I think it's fine to test the `by:` behavior.

Reply:
As far as I can see, it's not actually testing the `:by` behavior.  These args were added in https://github.com/rails/rails/pull/50490/commits/0868698bef7da1f7529f4013b5ed1f96acc413d9 simply to prevent flaky CI.  Also, by specifying `:by` here, we do not exercise the default value of `:by`.